sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala

```diff
@@ -70,11 +67,13 @@ object TPCDSQueryBenchmark extends SqlBasedBenchmark with Logging {
 
   def setupTables(dataLocation: String, createTempView: Boolean): Map[String, Long] = {
     tables.map { tableName =>
-      val df = spark.read.parquet(s"$dataLocation/$tableName")
       if (createTempView) {
-        df.createOrReplaceTempView(tableName)
+        spark.read.parquet(s"$dataLocation/$tableName").createOrReplaceTempView(tableName)
       } else {
-        df.write.saveAsTable(tableName)
+        spark.sql(s"DROP TABLE IF EXISTS $tableName")
+        spark.catalog.createTable(tableName, s"$dataLocation/$tableName", "parquet")
+        // Recover partitions but don't fail if a table is not partitioned.
+        Try(spark.sql(s"ALTER TABLE $tableName RECOVER PARTITIONS"))
```

Comment:
Should we add a debug or info log when it fails to recover partitions?

Reply:
Thanks, good idea. Added log in: https://github.com/apache/spark/pull/31218/commits/7eda4693cde7c2a1a5870e8dab3a6347678019bf